wiki_trac_rst_convert.py

```diff
@@ -38,42 +40,190 @@ def convert_content(text: str):
+    text = _tracwiki_subheading_to_rst_subheading(text)
+    text = _tracwiki_list_dedent(text)
+    text = _tracwiki_list_separate_from_paragraph(text)
 
     return text
 
 
-def _trac_rst_wiki_to_github_links(text: str):
+def _trac_to_github_wiki_links(text: str):
     """
-    Takes RST content with Trac wiki link directives
-    and coverts the directives to inline GitHub wiki links.
+    Takes content with Trac wiki link directives and coverts
+    the directives to inline GitHub wiki links.
     """
 
-    link_matchers =[re.compile(r) for r in [
+    link_matchers = [
+        # RST markup:
         ':trac:`wiki:(.+?)`',
-        '`wiki:(.+?)`:trac:'
-    ]]
+        '`wiki:(.+?)`:trac:',
+
+        # TracWiki markup:
+        '`\[wiki:"?([^ ]+?)"?]`:trac:',
+        '\[wiki:"?([^ ]+?)"?]',
+    ]
+
+    for link_re in link_matchers:
+        for title in _matches(link_re, text):
+            text = _sub(link_re, f'`<{_wiki_url(title)}>`_', text)
+
+    return text
+
+
+def _tracwiki_to_rst_links(text: str):
+    """
+    Takes TracWiki markup and converts its links to RST links.
+    """
+
+    url = '[a-z]+://[^ ]+'
+    link_text = '[^]]+'
+    link_re = f'\[({url}) ({link_text})]'
+
+    for url, link_text in _matches(link_re, text):
+        text = _sub(link_re, f'`{link_text} <{url}>`_', text)
+
+    return text
+
+
+def _tracwiki_wiki_link_with_text_to_github_links(text: str):
+    """
+    Takes TracWiki markup and converts its Wiki links which have
+    explicit link text into RST links.
+    If the link text is the same as the article name, generate a more
+    compact syntax.
+    """
+
+    title = '[^ ]+'
+    link_text = '[^]]+'
+
+    link_matchers = [
+        f'`\[wiki:({title}) ({link_text})]`:trac:',
+        f'\[wiki:({title}) ({link_text})]',
+    ]
 
     for link_re in link_matchers:
-        wiki_titles = re.findall(link_re, text)
-        for title in wiki_titles:
-            text = re.sub(
-                link_re,
-                rf'`<{_wiki_url(title)}>`_',
-                text,
-                1
-            )
+        for title, link_text in _matches(link_re, text):
+            if title == link_text:
+                text = _sub(link_re, f'`<{_wiki_url(title)}>`_', text)
+            else:
+                replacement = f'`{link_text} <{_wiki_url(title)}>`_'
+                text = _sub(link_re, replacement, text)
+
+    return text
+
+
+def _trac_ticket_links(text: str):
+    """
+    Replace Trac reference to ticket with an RST link to the ticket.
+    """
+
+    ticket_re = ':trac:`#([0-9]+)`'
+    for ticket in _matches(ticket_re, text):
+        text = _sub(
+            ticket_re,
+            f'`Trac #{ticket} <{TRAC_TICKET_PREFIX}{ticket}>`_',
+            text
+        )
+    return text
+
+
+def _tracwiki_heading_to_rst_heading(text: str):
+    """
+    Convert TracWiki headings (titles surrounded by "=") to RST (titles
+    underlined with "=").
+    """
```

Comment:
Maybe use some simple example in the docstring to explain how the conversion is done.

No need for all the details... we can have corner cases explained in the test code.

```suggestion
def _tracwiki_1st_heading_to_rst_heading(text: str):
    """
    Convert TracWiki  1st level headings to RST heading.
    
    TracWiki:
    
    = Some Top Heading =
    Content here
    
    RST conversion:
    
    Some Top Heading
    ----------------
    Content here
    """
```